### PR TITLE
[main] use proper URL for tracking when fetch is passed an empty string

### LINKED ISSUE #2164: [BUG] Fetch with empty string as first parameter does not include traceparent and does not successfully save dependency to Application Insights
**Description/Screenshot**
Now that #2149 is fixed, fetch calls with an empty string (`""`) succeed.  However, they do not add a `Traceparent` request header, and Application Insights rejects the dependency:
image

**Steps to Reproduce**

- OS/Browser: Windows 10/Chrome 117.0.5938.89
- SDK Version [e.g. 22]: Current snippet from https://learn.microsoft.com/en-us/azure/azure-monitor/app/javascript-sdk?tabs=javascriptwebsdkloaderscript.
- How you initialized the SDK: Javascript snippet

**Expected behavior**
Repro: https://jorupp.github.io/ai-post-test/with-ai.html and open dev tools, network tab.
Click "POST with URL" - as expected, it does a POST (which gets a 405), but it sends `Traceparent`.
Click "POST with empty string" - as expected, it does a POST (which gets a 405), but does not send `Traceparent`.

**Additional context**
I believe the issue is with https://github.com/microsoft/ApplicationInsights-JS/blob/3c8318bfad6cdd59b92ce87b28e86cc281c7cc2a/extensions/applicationinsights-dependencies-js/src/ajax.ts#L1187-L1191

When `input === ""`, it needs to set requestUrl to `window.location.href.split("#")[0]` (I can't find specific documentation that it should work this way, but browsers seem to include query string parameters on the request when the first arg is empty, though they do _not_ include hash parameters).

### PR DESCRIPTION
**Pull request by jorupp**
This allows AI to track the dependency and sets the Traceparent header on the request to the server (if otherwise allowed).

`Traceparent` is now set
image

Fully-populated dependency is sent to ApplicationInsights
image
image

Data is displayed in Application Insights
image

Fixes #2164

### CODE REVIEW

File: extensions/applicationinsights-dependencies-js/Tests/Unit/src/ajax.tests.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿import { SinonStub } from "sinon";
+import { SinonStub } from "sinon";
```
Comment (jorupp) [inline extensions/applicationinsights-dependencies-js/Tests/Unit/src/ajax.tests.ts:1]: hmm.. this change wasn't in what I staged, and my editor is telling me it's not actually changed.  I'm guessing this is a CR/LF issue I need to sort out.

Comment (jorupp) [reply extensions/applicationinsights-dependencies-js/Tests/Unit/src/ajax.tests.ts]: ok, got it fixed by using `git` instead of VSCode to do the interactive add to ignore that hunk.

File: extensions/applicationinsights-dependencies-js/src/ajax.ts [hunk 2]
```diff
@@ -1186,6 +1186,9 @@ export class AjaxMonitor extends BaseTelemetryPlugin implements IDependenciesPlu
 
                 if (input instanceof Request) {
                     ajaxData.requestUrl = input ? input.url : "";
+                } else if (input === "" ) {
+                    const location = getLocation();
```
Comment (jorupp) [inline extensions/applicationinsights-dependencies-js/src/ajax.ts:1190]: this seemed to be the way other code was accessing `window.location`

File: extensions/applicationinsights-dependencies-js/src/ajax.ts [hunk 3]
```diff
@@ -1186,6 +1186,9 @@ export class AjaxMonitor extends BaseTelemetryPlugin implements IDependenciesPlu
...
+                    ajaxData.requestUrl = location?.href?.split("#")[0] || input;
```
Comment (MSNev) [inline extensions/applicationinsights-dependencies-js/src/ajax.ts:1191]: We tend to not use the TypeScript `?.` operator as this causes a lot of JavaScript to get generated and we are always attempting to keep the size of the generated JS code to a minimum.

Suggest maybe something like the following so that we handle the case where input.url === ""
```ts
let requestUrl: string;
if (input instanceof Request) {
    requestUrl = (input||{}).url || "";
} else {
    requestUrl = input;
}

// Handle case where either the input (as a string) or the input.url is an empty string
if (requestUrl === "") {
    const location = getLocation();
    if (location && location.href) {
        requestUrl = strSplit(location.href, "#")[0];
    }
}

ajaxData.requestUrl = requestUrl;
```

Where `strSplit` is imported from `@nevware21/ts-utils`

File: extensions/applicationinsights-dependencies-js/src/ajax.ts (hunk 2 above)
Comment (jorupp) [reply extensions/applicationinsights-dependencies-js/src/ajax.ts:1190]: well, if i wrote the code, I wouldn't have used `""`, but that's how NextJS 13's app router handles server-actions - I brought it up in [a discussions there](https://github.com/vercel/next.js/discussions/55405#discussioncomment-7006185), but after digging further into the code that appeared to be causing the issue (<https://github.com/vercel/next.js/blob/82eb6e6f4955dc90752a4875ccf86b7ac0c8d03a/packages/next/src/client/components/router-reducer/reducers/server-action-reducer.ts#L58>), it didn't seem that what the Next code was doing was actually contrary to how `fetch` is documented (or actually) works, so I figured the fix was better on this side in case there were other apps/libraries taking advantage of that behavior for `""` with `fetch`.

File: extensions/applicationinsights-dependencies-js/src/ajax.ts (hunk 3 above)
Comment (jorupp) [reply extensions/applicationinsights-dependencies-js/src/ajax.ts:1191]: ah, so not a "does it work" thing, but "how much JS gets generated" - got it.

Comment (jorupp) [reply extensions/applicationinsights-dependencies-js/src/ajax.ts:1191]: oh, that's a good point too re: "what if `input.url === ""`"

Comment (jorupp) [reply extensions/applicationinsights-dependencies-js/src/ajax.ts:1191]: `new Request("")` appears to be setting `url` to the current URL itself, but it doesn't hurt to handle that here.

Comment (jorupp) [reply extensions/applicationinsights-dependencies-js/src/ajax.ts]: update pushed

Comment (MSNev) [reply extensions/applicationinsights-dependencies-js/src/ajax.ts]: > ah, so not a "does it work" thing, but "how much JS gets generated" - got it.

Will need to create a work item to track those down and remove ☹️

**Review by MSNev — APPROVED** [review]

**Comment by MSNev** [comment]
@microsoft-github-policy-service

**Comment by MSNev** [comment]
@microsoft-github-policy-service rerun

**Comment by MSNev** [comment]
Going to close and re-open to force the CLA to rerun
